openexr-rs/src/tiledinputfile.rs

```diff
@@ -0,0 +1,970 @@
+        let c_file_name = CString::new(
+            file_name
+                .as_ref()
+                .to_str()
+                .expect("Invalid bytes in filename"),
+        )
+        .expect("Internal null bytes in filename");
+
+        unsafe {
+            sys::Imf_TiledInputFile_ctor(
+                &mut ptr,
+                c_file_name.as_ptr(),
+                num_threads,
+            )
+            .into_result()?;
+
+            // TODO: Validate if OpenEXR will always return a valid tiled input
+            // file.
+            if ptr.is_null() {
+                panic!("Received null ptr from sys::Imf_TiledInputFile_ctor");
+            }
+        }
+
+        Ok(Self { inner: ptr })
+    }
+
+    /// Access to the file path
+    ///
+    pub fn file_name(&self) -> PathBuf {
+        let mut ptr = std::ptr::null();
+
+        unsafe {
+            sys::Imf_TiledInputFile_fileName(self.inner, &mut ptr)
+                .into_result()
+                .unwrap();
+
+            // TODO: Validate if OpenEXR will always return a valid string, or
+            // if we need to return a Result<PathBuf, Error>.
+            if ptr.is_null() {
+                panic!(
+                    "Received null ptr from sys::Imf_TiledInputFile_fileName"
+                );
+            }
+
+            let file_name = CStr::from_ptr(ptr);
+
+            PathBuf::from(
+                file_name.to_str().expect("Invalid bytes in filename"),
+            )
+        }
+    }
+
+    /// Access to the file header
+    ///
+    pub fn header<'a>(&'a self) -> HeaderRef<'a> {
+        let mut ptr = std::ptr::null();
+
+        unsafe {
+            sys::Imf_TiledInputFile_header(self.inner, &mut ptr)
+                .into_result()
+                .unwrap();
+
+            // TODO: Validate if OpenEXR will always return a valid header, or
+            // if we need to return a Result<Header, Error>.
+            if ptr.is_null() {
+                panic!("Received null ptr from sys::Imf_TiledInputFile_header");
+            }
+        }
+
+        HeaderRef::new(ptr)
+    }
+
+    /// Access to the file format version
+    ///
+    pub fn version(&self) -> crate::Version {
+        let mut version = 0;
+
+        unsafe {
+            sys::Imf_TiledInputFile_version(self.inner, &mut version)
+                .into_result()
+                .unwrap();
+        }
+
+        crate::Version::from_c_int(version)
+    }
+
+    /// Set the current frame buffer -- copies the FrameBuffer
+    /// object into the TiledInputFile object.
+    ///
+    /// The current frame buffer is the destination for the pixel
+    /// data read from the file.  The current frame buffer must be
+    /// set at least once before read_tile() is called.
+    /// The current frame buffer can be changed after each call
+    /// to read_tile().
+    ///
+    /// # Errors
+    ///
+    /// Returns an error if the sub-sampling factors of the input file
```

Comment:
We can come back and hit these later, but it would be nice to spell out what errors to expect and in what context, e.g. https://github.com/vfx-rs/openexr-bind/blob/e4052f0d2a2ab5cfb87822e7b2468effeae26337/openexr-rs/src/header.rs#L89